### DISCUSSION TITLE
Docker: How do you edit superuser password and edit INI config file for Murmur?

### DISCUSSION THREAD

**Discussion by trymeouteh** [mumble-voip/mumble-docker#16] [Q&A] [answered]
I am a newbie when it comes to using docker and I cannot figure out how to configure my Murmur docker server. I was able to pull the image using the instructions on github and run the server in the terminal, but I am unable to configure the server and change the superuser password.

Any help and tips will be most appreciated

**Comment by azlux** [comment]
Please refer to https://github.com/mumble-voip/mumble-docker#configuration for config values into the INI file
or
https://github.com/mumble-voip/mumble-docker#additional-variables for specials values for mumble-server

So you can set the password with environment variable ( with MUMBLE_SUPERUSER_PASSWORD )

If you want a custom ini file, check the MUMBLE_CUSTOM_CONFIG_FILE variable. It's allow you to have you own file (you need to mount it, see https://docs.docker.com/storage/bind-mounts/ )

You have one example (with docker-compose) into the wiki : https://github.com/mumble-voip/mumble-docker/wiki/docker_example_mumble_mumo_postgres

**Reply by trymeouteh** [reply]
For whatever reason when I add the _-e "MUMBLE_CUSTOM_CONFIG_FILE=/data/murmur.ini"_ into the run command the mumble server will not launch. I did create the volume before hand and placed the murmur.ini file inside the volume /data/ folder with the database path set in the INI file to _/data/murmur.sqlite_


`docker run -e "MUMBLE_SUPERUSER_PASSWORD=superUSER" -e "MUMBLE_CUSTOM_CONFIG_FILE=/data/murmur.ini" --name mumbleCONTAINER --publish 64738:64738/tcp --publish 64738:64738/udp --volume mumbleVOLUME:/data mumblevoip/mumble-server`

**Reply by Krzmbrzl** [reply]
You should provide the log, otherwise we won't be able to help...

**Reply by trymeouteh** [reply]
Here are the logs when I run the container.

```
{"log":"Using manually specified config file at /data/mumble-server.ini\n","stream":"stdout","time":"2022-09-17T02:06:33.190756507Z"}
{"log":"All MUMBLE_CONFIG variables will be ignored\n","stream":"stdout","time":"2022-09-17T02:06:33.190779356Z"}
{"log":"\u003cX\u003e2022-09-17 02:06:33.196 SSL: OpenSSL version is 'OpenSSL 1.1.1f  31 Mar 2020'\n","stream":"stderr","time":"2022-09-17T02:06:33.196879544Z"}
{"log":"\u003cW\u003e2022-09-17 02:06:33.196 Initializing settings from /data/mumble-server.ini (basepath /data)\n","stream":"stderr","time":"2022-09-17T02:06:33.196902512Z"}
{"log":"\u003cW\u003e2022-09-17 02:06:33.375 MetaParams: TLS cipher preference is \"TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:AES256-SHA:AES128-SHA\"\n","stream":"stderr","time":"2022-09-17T02:06:33.375218066Z"}
{"log":"\u003cF\u003e2022-09-17 02:06:33.395 ServerDB: Failed initialization: unable to open database file Error opening database\n","stream":"stderr","time":"2022-09-17T02:06:33.39559482Z"}
```

**Reply by Krzmbrzl** [reply]
It seems to use your INI file as expected. What did you specify for the `database` config in your INI?

**Reply by trymeouteh** [reply]
This is how I set the database in the INI file. There is not database in the volume but it should generate a database file.
`database=mumble-server.sqlite`

**Reply by Krzmbrzl** [reply]
The path you specify here should point to `/data/`. As it stands the server executable will try to create the database relative to where the executable resides andmost likely it doesn't have write access there

**Reply by trymeouteh** [reply]
I changed the `database` value in the INI to `database=/data/mumble-server.sqlite`. Not sure how to add write access to this directory in docker. These are the commands I used to create the volume and run the docker container...

```
docker volume create MYVOLUME
docker run -e "MUMBLE_SUPERUSER_PASSWORD=MYPASSWORD" -e "MUMBLE_CUSTOM_CONFIG_FILE=/data/mumble-server.ini" --name MYCONTAINER --publish 44999:64738/tcp --publish 44999:64738/udp --volume MYVOLUME:/data mumblevoip/mumble-server
```

I think I am getting close!

**Reply by Krzmbrzl** [reply]
> The path you specify here should point to /data/

...

**Reply by trymeouteh** [reply]
Do I add /data/ to the INI file, or to the run command?

**Reply by Krzmbrzl** [reply]
> I changed the database value in the INI to database=/data/mumble-server.sqlite

Exactly like that. Iirc though before your edit your message only stated `database=mumble-server.sqlite` - thud my comment. 

I'm not sure if your volume mapping is correct. Maybe @azlux can have a look at that - I don't have a clue how that's supposed to work ^^

**Reply by trymeouteh** [reply]
Fair enough, I did change the message since I meant to post `database=/data/mumble-server.sqlite`

**Reply by trymeouteh** [reply]
> You have one example (with docker-compose) into the wiki : https://github.com/mumble-voip/mumble-docker/wiki/docker_example_mumble_mumo_postgres

Are there any example with using an INI file for the settings inside a docker volume with the sqlite database being inside the docker volume also?

**Comment by azlux** [comment, marked as answer]
Your example should look like : 
```
  mumble-server:
    image: mumblevoip/mumble-server
    container_name: mumble-server
    hostname: mumble-server
    restart: on-failure
    environment:
      MUMBLE_CUSTOM_CONFIG_FILE: "/data/mumble-server.ini"
    volume:
      - <your/path/or/volume/config.ini>:/data/mumble-server.ini
      - <your/path/or/volume/db.sqlite>:/data/mumble-database.sql
    ports:
      - 64738:64738
      - 64738:64738/udp
```
It's easier if you use a path instead of volume. Your file need the have the write access for uid 1000

**Reply by trymeouteh** [reply]
How would I convert all of this into a docker run command? I would like to have the option to run multiple docker mumble servers at once also, of course the other mumble servers will run on different ports.

**Reply by azlux** [reply]
can we close this discussion ? Or you still need help ?
